### PR TITLE
Add redux stuff

### PR DESCRIPTION
**Pull request by reald002**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -16,7 +16,9 @@
         "@types/react-dom": "^17.0.0",
```
Comment (AlexButov) [inline package-lock.json:16]: don't send this file in PRs

File: src/index.tsx [hunk 2]
```diff
@@ -2,10 +2,14 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
+import { Provider } from 'react-redux';
+import { store } from './redux';
```
Comment (AlexButov) [inline src/index.tsx:4]: incorrect import sequence

File: src/redux/actions.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ADD_TODO, TOGGLE_TODO, REMOVE_TODO } from './types';
+import {IAddTodo, IRemoveTodo, IToggleTodo} from '../core/interfaces';
```
Comment (AlexButov) [inline src/redux/actions.tsx]: spaces

File: src/redux/actions.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+export const addTodo = (text: string): IAddTodo => {
+  return {
+    type: ADD_TODO,
+    text
+  };
+};
+
+export const removeTodo = (id: string | number): IRemoveTodo => {
```
Comment (AlexButov) [inline src/redux/actions.tsx]: why your id can be number or string?

File: src/redux/actions.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  return {
+    type: REMOVE_TODO,
+    id
+  };
+};
+
+export const toggleTodo = (id: string | number): IToggleTodo => {
```
Comment (AlexButov) [inline src/redux/actions.tsx]: the same question

File: yarn.lock [hunk 6]
```diff
@@ -1131,7 +1131,7 @@
     "core-js-pure" "^3.0.0"
```
Comment (AlexButov) [inline yarn.lock:1131]: don't send this file in PRs

**Review by AlexButov — CHANGES_REQUESTED** [review]

File: src/redux/actions.tsx (hunk 4 above)
Comment (reald002) [reply src/redux/actions.tsx]: @AlexButov 
For now it's Date.now() which is number, but it also may be id generated from mongodb which is string. So I'm not sure should it be number or string

**Pull request merged**